lib/VMwareWebService/broker_timeout.rb

```diff
@@ -0,0 +1,64 @@
+module Timeout
```

Comment:
Mostly for the sake of posterity / myself in a few months: I disagree with the characterisation of upstream timeout as improper, and this as a "fix" -- it's my position that timeout is fine as is, and the problem we're seeing is deeper in the `Thread#raise`-vs-`kill+join` interaction (which would make this a workaround).

But regardless of precisely where we place the blame, this is an effective means of making it not happen, and the above is a strong description of the behaviour we're seeing / trying to avoid. :+1: